token-service/src/main/java/org/p2p/token/service/repository/price/TokenPriceDatabaseRepository.kt

```diff
@@ -0,0 +1,39 @@
+package org.p2p.token.service.repository.price
+
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import org.p2p.core.dispatchers.CoroutineDispatchers
+import org.p2p.token.service.database.TokenPriceDao
+import org.p2p.token.service.database.mapper.TokenServiceDatabaseMapper
+import org.p2p.token.service.model.TokenServiceNetwork
+import org.p2p.token.service.model.TokenServicePrice
+
+internal class TokenPriceDatabaseRepository(
+    private val tokenPriceDao: TokenPriceDao,
+    private val converter: TokenServiceDatabaseMapper,
+    coroutineDispatchers: CoroutineDispatchers
+) : TokenPriceLocalRepository, CoroutineScope {
+
+    override val coroutineContext: CoroutineContext = coroutineDispatchers.io
+    override fun setTokensPrice(prices: List<TokenServicePrice>) {
```

Comment:
why isn't it a suspend function?

Reply:
Nonetheless, this function is essentially a suspend function because it's entirely wrapped in the launch block.
If you want to maintain specific dispatcher for this specific db operation you can just use `withContext(YOUR_DB_DISPATCHER)`.

Reply:
I meant, that in this way we make it async, and don't suspend launch thread from presenter